src/main/java/fi/aalto/cs/apluscourses/intellij/model/task/ProjectTreePresenter.java

```diff
@@ -1,17 +1,29 @@
 package fi.aalto.cs.apluscourses.intellij.model.task;
 
+import com.intellij.openapi.project.Project;
 import fi.aalto.cs.apluscourses.ui.ideactivities.ComponentDatabase;
 import fi.aalto.cs.apluscourses.ui.ideactivities.GenericHighlighter;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public class ProjectTreePresenter extends IntelliJComponentPresenterBase {
-  public ProjectTreePresenter(@NotNull String instruction, @NotNull String info) {
-    super(instruction, info);
+  public ProjectTreePresenter(@NotNull String instruction,
+                              @NotNull String info,
+                              @NotNull Project project) {
+    super(instruction, info, project);
   }
 
   @Override
   protected @Nullable GenericHighlighter getHighlighter() {
-    return new GenericHighlighter(ComponentDatabase.getProjectPane());
+    var component = ComponentDatabase.getProjectPane();
+    if (component == null && tryToShow()) {
```

Comment:
I see this method as a "getter" but it apparently have quite a visible side effect: it displays the project tree. I'd strongly suggest reconsidering this.

Reply:
oops i left that there accidentally